### PR TITLE
CLOUDP-77155: Fix fields marshal/unmarshal for automation config

### PR DESCRIPTION
**Pull request by svagner**
## Proposed changes

- Add setParameter to args26 struct
- set field' NumCores in Process struct tag as omitempty

_Jira ticket:_ CLOUDP-77155

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: opsmngr/automation_config.go [hunk 1]
```diff
@@ -344,7 +345,7 @@ type Process struct {
 	LogRotate                                  *LogRotate         `json:"logRotate,omitempty"`
 	ManualMode                                 bool               `json:"manualMode"`
 	Name                                       string             `json:"name,omitempty"`
-	NumCores                                   int                `json:"numCores"`
+	NumCores                                   int                `json:"numCores,omitempty"`
```
Comment (gssbzn) [inline opsmngr/automation_config.go:348]: if zero is not a valid value this is ok, if not we may need to change this to a pointer

File: go.mod [hunk 2]
```diff
@@ -11,6 +11,7 @@ require (
 	github.com/xdg/stringprep v1.0.0 // indirect
 	go.mongodb.org/atlas v0.5.1-0.20201007214134-b315fe7503d2
 	golang.org/x/text v0.3.3 // indirect
+	golang.org/x/tools v0.0.0-20201116182000-1d699438d2cf // indirect
```
Comment (gssbzn) [inline go.mod:14]: [q] was this and the changes to the go.sum file necessary? can you please do `go mod tidy`?

**Review by gssbzn — CHANGES_REQUESTED** [review]
Hi and thanks for the fix, major thing here is to check if the changes to the dependencies was necessary as I don't think they are

File: opsmngr/automation_config.go (hunk 1 above)
Comment (svagner) [reply opsmngr/automation_config.go:348]: As I understood `0` is a valid value but it won’t be considered within mongod - https://github.com/mongodb/mongo/blob/2931f3c764f8e093ae31adb4b2b62c1ce01d7421/src/mongo/executor/task_executor_pool.cpp#L51

File: go.mod (hunk 2 above)
Comment (svagner) [reply go.mod]: Yeah. Looks like I forgot to run it. Sorry. It was fixed 👍

**Review by gssbzn — APPROVED** [review]
Thanks for these fixes

**Pull request merged**